runtime_lib/controller/main.cpp

```diff
@@ -820,6 +820,48 @@ void xaie_l2_dma_init(int col) {
   xaie::out32(xaie::getTileAddr(col, 0) + 0x00033008, 0xFF);
 }
 
+#ifdef ARM_CONTROLLER
+
+// Defining the NPI base and registers we use to reset the array
+uint64_t npi_base = 0xF70A0000UL;
+#define NPI_MASK_REG 0x0
```

Comment:
Since you are lucky enough to use C++ instead of C :-)
```suggestion
constexpr uint64_t npi_base = 0xF70A0000UL;
constexpr auto NPI_MASK_REG = 0x0;
```
?